doc/usage/configuration.rst

```diff
@@ -1191,6 +1191,11 @@ that use Sphinx's HTMLWriter class.
 
    .. versionadded:: 3.5
 
+.. confval:: html_signaturereturn_icon
```

Comment:
I still don't understand why this is needed. Do you want to change the character of the allow?
Adding a new configuration increases the cost of maintenance. So I need to know this is really needed.

Note:

* Which is better `html_signaturereturn_icon` and `html_signature_return_icon`.
* If we'll determine to add this, testcase is needed.
* If we'll determine to add this, it will be released as 4.x (maybe 4.1).

Reply:
I've fixed the typo.

Yes, I would like to be able to replace the arrow character.

My reason for this is that I would like to be able to display the function/method signature without the type hints. The type hints should however be displayed when the user hovers over an annotation icon with the mouse. This should work for both the parameter type hints and the return type hints. I would like to use consistent icons for this. For example https://fontawesome.com/icons/question-circle?style=regular for the parameter hint and https://fontawesome.com/icons/arrow-circle-right?style=regular for the return hint.

I find long signatures with type hints confusing. An extreme example from Sphinx (in `application.py` in `Sphinx`):

```python
    def __init__(self, srcdir: str, confdir: Optional[str], outdir: str, doctreedir: str,
                 buildername: str, confoverrides: Dict = None,
                 status: IO = sys.stdout, warning: IO = sys.stderr,
                 freshenv: bool = False, warningiserror: bool = False, tags: List[str] = None,
                 verbosity: int = 0, parallel: int = 0, keep_going: bool = False) -> None:
```

IMHO it's hard to see at first glance what is a parameter name, what is a type annotation and what is a default value (even **with** syntax highlighting). Without the type annotations this look much better:

```python
    def __init__(self, srcdir, confdir, outdir, doctreedir, buildername, confoverrides,
                 status=sys.stdout, warning=sys.stderr, freshenv=False, warningiserror=False,
                 tags=None, verbosity=0, parallel=0, keep_going=False):
```

But I don't want to drop the type annotations completely, because they're useful when you are writing the function call. Fortunately on an HTML page we can have both: short and clear signatures **and** access to the type hints: By displaying the type hints on demand via a popup/tooltip.

I realize that this currently isn't possible for the parameter hint, but this patch makes it possible for the return hint.

I chose `html_signaturereturn_icon` for that name, because this matches the name for `html_permalinks_icon`.

However if we would introduce an icon for the parameter annotation in the future, `html_signature_return_icon` would make more sense, since we might have `html_signature_param_icon` too.

I'll start working on a testcase.

OK, I'll add a `.. versionadded:: 4.1` to the documentation.

Reply:
Thank you for your explanation. I understand it is worthy for you. But, I don't think it's also worthy for other users. Is it possible to change the icon by CSS hack or JS?